nms/README.md

```diff
@@ -1,14 +1,141 @@
 
-# Testing
+## Project Layout
 
-  yarn run eslint ./
-  flow
-  yarn run test
+There are two main directories, and `package.json` files.
 
-# Testing Coverage
+### Main Application Directory
+In `magma/nms/packages/magmalte`, the main NMS application lives.
+Here, `package.json` specifies NMS application dependencies.
+
+### Dev Workflows Directory
+In `magma/nms`, this is to be merged with the main application directory.
+Here, `package.json` specifies dependencies necessary for development workflows and testing.
+Specifically, triggering unit tests, e2e tests, eslint, and flow tests are done from this directory.
+
+## Running Local Dev NMS
+> NOTE: This guide is written for development directly using Docker.
+> You may need to make adjustments if using MiniKube.
+
+### Pre-requisites
+The NMS requires a working connection to the Magma Orc8r to function.
+For local development, bring up the Orc8r docker containers.
+See `magma/orc8r/cloud/docker/docker-compose.yml`
+To ensure that NMS has access to metrics and graphing functionality, bring up the Prometheus and Grafana containers.
+See `magma/orc8r/cloud/docker/docker-compose.metrics.yml`
+
+### Build NMS Docker Image
+Run the following command from `magma/nms/packages/magmalte`
+```
+docker-compose build magmalte
+```
+
+### Run NMS Containers
+Run the following command from `magma/nms/packages/magmalte`
+```
+docker-compose up -d
+```
+
+## Checking NMS Logs
+Run the following from `magma/nms/packages/magmalte`
+```
+docker-compose logs -f magmalte
+```
+Depending on the issues you run into, you may need to check the logs of other containers, both NMS and Orc8r related.
+
+## Testing
+
+### Install Dependencies
+Running eslint and flow tests requires installing dependencies.
+Run the following command from `magma/nms` to install these dependencies:
+```
+yarn
+```
+
+### Eslint
+Run from `magma/nms`
+```
+yarn run eslint ./
+```
+
+### Flow
+Run from `magma/nms`
+```
+flow
+```
+
+### Unit Tests
+Run from `magma/nms`
+```
+yarn run test
+```
+
+### Testing Coverage
 
 Run `yarn test --coverage`
+
+## Updating NMS for API Changes
+
+### How to re-generate MagmaAPIBindings.js
+1. Place an up-to-date `swagger.yml` in `magma/nms/packages/magmalte`
+   You can get the full `swagger.yml` at `{orc8r domain}/swagger/v1/spec`
+   (e.g. `https://localhost:9443/swagger/v1/spec`)
+2. Run `scripts/generateAPIFromSwagger.sh` from `magma/nms/packages/magmalte`
+
+
+## Accessing Local Dev NMS
+
+### Multitenancy and Organizations
+
+Multitenancy is supported in the Magma NMS. Each tenant is called an "organization".
+Each organization owns a subset of the networks provisioned on Orchestrator, and the special `master` organization administrates organizations in the system.
+
+Users in organizations log into the NMS using a subdomain that matches their organization name.
+For example, users of a *NewOrg* organization in the NMS would access the NMS using http://NewOrg.localhost:8081/nms
```

Comment:
to make it clear that NewOrg is going to be replaced maybe put it in <>
http://\<NEWORG\>.localhost:8081/nms